removed moment as a dependency

reubenae opened the pull request:
# Title

## Description of what's changing

Removed moment as a dependency to cut down package size

## What else might be impacted?

...

## Checklist

[ ] Tests are written and maintain or improve code coverage
[ ] I've tested this in my application using `yarn link` (if applicable)
[ ] You consent to and are confident this change can be released with no regression

reubenae commented on package.json:
```diff
@@ -62,6 +61,7 @@
     "jest-cli": "^26.1.0",
     "jest-coverage-badges": "^1.1.2",
     "lint-staged": "^10.2.11",
+    "moment": "^2.28.0",
```
Maybe slightly controversial but I figured there was no better way of testing this change than by keeping the test the same. So I've moved moment to be a dev dependency (so bundle size is minimal) but we still get to keep our good unit test. And other tests in the future can take advantage of moment.

reubenae commented:
@skodamarthi this is ready for review now

skodamarthi commented on src/js/common/browserInfoHelper.js:
```diff
@@ -68,7 +67,22 @@ const validateAndGetScreenDetail = (value) => {
+    const hourIndexStart = 0;
+    const hourLength = 3;
+    const minuteIndexStart = 3;
+    const minuteLength = 2;
```
Didn't quite understand this logic. Is this the standard logic for parsing to UTC timezone?

skodamarthi replied:
Also, it looks like we are just removing moment from here to reduce the bundle size but will still continue using moment for unit tests. Is this a standard practice? I find it a bit confusing to use it in unit tests but not in the actual code.

reubenae replied:
Yes, this method now just essentially gets the right bits from the string (which is in a standard format that does not change) and reconstructs it into the format we require (with the `:`)

It is quite normal to have dependencies which are only used during testing and development. It is a bit unusual for moment.js to be one, but I think it makes perfect sense. What is the alternative? Essentially to rewrite this method to tests itself, which is very circular and gives me less confidence than testing against the gold standard of moment.js. 

The other benefit is that it ensures the test is environment agnostic, e.g. it doesn't care which timezone the test is being run in. To test without moment we would either tie ourselves to the environment the test is run in (e.g. is it UTC+1 or UTC + 7?) *or* mock the global variable, which makes the test quite open to being wrong, in my opinion.

So I would say there is significant upside to keeping moment as a devDependency (quality and durability of the test) with no/very limited downside (developers have to install the package to run the tests)

reubenae commented:
Moment is gone completely now. And I rebased since we have had a decent number of contributions since I branched. I also found a slightly cleaner way to pick out the bits of the formatted string we want... and added a comment besides to make it clearer when we look back in a years time. Ready for review and merge when you have some time @skodamarthi

skodamarthi commented on src/js/common/browserInfoHelper.js:
```diff
@@ -68,7 +67,19 @@ const validateAndGetScreenDetail = (value) => {
+    // nowUTC is in format like "Wed Sep 30 2020 23:11:02 GMT+0100 (British Summer Time)"
+    // To format the offset, we pick the relevant characters based on their position
+    // and reformat with a ":"
+    const signCharIndex = 28;
```
I was thinking figuring out position 28, 29 etc; from such long string is difficult. How about splitting the string to bare minimum required (i.e +0100)  and applying the logic to fetch hour and minute offset from that?

Eg: const nowUTC = new Date().toString().split('GMT')[1].split(' ')[0]; // gives "+0100"

honsq90 replied:
if we're just reading the timestamp from the current browser session, it's possible to use [getTimezoneOffset](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset)?

```javascript
const offset = new Date().getTimezoneOffset() * -1;
const offsetHours = `${Math.floor(offset / 60)}`.padStart(2, 0);
const offsetMinutes = `${Math.abs(offset % 60)}`.padStart(2, 0);
const sign = offset >= 0 ? '+' : '-'
const offsetString = `UTC${sign}${offsetHours}:${offsetMinutes}`

```

reubenae replied:
> if we're just reading the timestamp from the current browser session, it's possible to use [getTimezoneOffset](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset)?
> 
> ```js
> const offset = new Date().getTimezoneOffset() * -1;
> const offsetHours = `${Math.floor(offset / 60)}`.padStart(2, 0);
> const offsetMinutes = `${Math.abs(offset % 60)}`.padStart(2, 0);
> const sign = offset >= 0 ? '+' : '-'
> const offsetString = `UTC${sign}${offsetHours}:${offsetMinutes}`
> ```

While this would work it looks a good deal more complicated than the current version in the PR, which is based on @skodamarthi 's suggestion.
